### PR TITLE
ROX-24429: integrate exposure analysis into roxctl

### PR DESCRIPTION
**Pull request by shireenf-ibm**
### Description

roxctl integration with np-guard enhancements (delayed from ACS 4.6)
in this PR integrating exposure analysis into rxctl netpol connectivity map command
i.e. added the new flag --exposure; 
 `roxctl netpol connectivity map --exposure`

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed (not needed)
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed (not needed)

### Testing and quality

- [ ] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [ ] CI results are inspected

#### Automated testing

- [x] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [ ] modified existing tests
- added bats tests
#### How I validated my change

- this is an integration with the new version of np-guard/netpol-analyzer,
- added the new flag and new relevant unit-test and bats-tests so integrated successfully

### CODE REVIEW

File: roxctl/netpol/connectivity/map/exposure_analysis.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Exposure analysis
+
+The goal of exposure analysis, is to provide additional information of potential permitted connectivity to the output of `roxctl netpol connectivity map` command.
+
+The report can give hints to where network policies may be tightened, or help validate that no unexpected exposure is present due to policies misconfiguration.
+
+Exposure-analysis is supported with `txt, md, csv, json, dot` output formats
```
Comment (vikin91) [inline roxctl/netpol/connectivity/map/exposure_analysis.md:7]:
```suggestion
Exposure-analysis is supported with `txt, md, csv, json, dot` output formats.
```

File: roxctl/netpol/connectivity/map/exposure_analysis.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# Exposure analysis
+
+The goal of exposure analysis, is to provide additional information of potential permitted connectivity to the output of `roxctl netpol connectivity map` command.
+
+The report can give hints to where network policies may be tightened, or help validate that no unexpected exposure is present due to policies misconfiguration.
```
Comment (vikin91) [inline roxctl/netpol/connectivity/map/exposure_analysis.md:5]:
```suggestion
The report contains hints to where network policies may be tightened, or help validate that no unexpected exposure is present due to policies misconfiguration.
```

File: roxctl/netpol/connectivity/map/exposure_analysis.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+![graph](exposure-graph-example.svg)
+
+#### Understanding the graph
+
+- `gold` edges are edges produced by the original connectivity map report
+- `blue` nodes and labels represent input workloads.
+- `black` frames and labels represent namespaces of input manifests
+- `red` nodes and labels are used to represent: - representative entities which are matched by policy rules, but do not exist in the input manifests\
+                                                - external IP addresses
```
Comment (vikin91) [inline roxctl/netpol/connectivity/map/exposure_analysis.md:62]: Why there are so many spaces at the beginning of this line? Could you check what happened here?

**Review by vikin91 — COMMENTED** [review]
Hi @shireenf-ibm , thanks for submitting this PR. I looked at the code and found only minor corrections.

As the CI is not running on this PR, I would need to duplicate is and let the CI run. A solution to that problem would be to get your team a Github user that is part of the Stackrox organisation - that would make future PR reviews easier, as that user could run all CI on their own. I will check whether there is a procedure for adding such user and will let you know.

Additionally, we would like to add you as codeowners to the parts of the code that rely on the output of NP-Guard. This would help other colleagues (and our CI) identify whom to contact if there are any failing tests or issues. Would that be okay?

**Comment by shireenf-ibm** [comment]
> Hi @shireenf-ibm , thanks for submitting this PR. I looked at the code and found only minor corrections.
> 
> As the CI is not running on this PR, I would need to duplicate is and let the CI run. A solution to that problem would be to get your team a Github user that is part of the Stackrox organisation - that would make future PR reviews easier, as that user could run all CI on their own. I will check whether there is a procedure for adding such user and will let you know.
> 
> Additionally, we would like to add you as codeowners to the parts of the code that rely on the output of NP-Guard. This would help other colleagues (and our CI) identify whom to contact if there are any failing tests or issues. Would that be okay?

Hi @vikin91 ,
thanks for the review;
I appreciate checking if there is a procedure for adding us as Github user, it would be very helpful.
And of course it would be okay to add me as code-owners of the relevant code parts; will be glad to help

**Comment by vikin91** [comment]
@shireenf-ibm we have created a team [IBM-Research](https://github.com/orgs/stackrox/teams/ibm-research) and added it as a member of Stackrox. You, Adi and Ziv are members in this team. Could we please check if you can now open a PR directly against the stackrox repo?

**Comment by vikin91** [comment]
> sure, what should I do now?

Checkout the stackrox repo directly (not into a fork), open a new branch against master, push changes from this PR to that branch and open a new PR. If that works well and CI is running, then we can review and discuss in this new PR and this can be closed.

**Pull request closed without merging**